src/apps/js_generic/js_generic.cpp

```diff
@@ -290,7 +290,7 @@ namespace ccfapp
 
       auto scripts = tx.get_view(this->network.app_scripts);
       scripts->foreach([&out](const auto& key, const auto&) {
-        out.methods.push_back(key);
+        out.endpoints.push_back({"POST", key});
```

Comment:
This isn't correct. We currently have keys like `GET log/private`, and use these for dispatch (see ~line 171 in `js_generic.cpp`). For backwards compatibility, if we don't a `verb` prefix then we'll actually match anything (ie - if we install a handler at just `foo/bar`, it'll accept `GET foo/bar`, `DELETE foo/bar`, etc).

This backwards compatibility is probably pointless since our js/lua support is still barebones, so we should probably instead require that we have a verb-prefix when we add things to the scripts table. Then here we can split on `" "` (or a better delimiter) to find the supported verb and path.

Reply:
Thanks for the heads up!